addons/portal/static/src/js/portal.js

```diff
@@ -46,6 +49,50 @@ publicWidget.registry.portalDetails = publicWidget.Widget.extend({
     _onCountryChange: function () {
         this._adaptAddressForm();
     },
+
+    /**
+     * @private
+     * @param {Event} ev
+     */
+    _onEditProfilePicClick: function (ev) {
+        ev.preventDefault();
+        ev.currentTarget.closest("form").querySelector(".o_portal_file_upload").click();
+    },
+
+    /**
+     * @private
+     * @param {Event} ev
+     */
+    _onFileUploadChange: function (ev) {
+        if (!ev.currentTarget.files.length) {
+            return;
+        }
+        const formEl = ev.currentTarget.closest("form");
+        const reader = new window.FileReader();
+        reader.readAsDataURL(ev.currentTarget.files[0]);
+        reader.onload = function (ev) {
+            formEl.querySelector(".o_wportal_avatar_img").src = ev.target.result;
+        };
```

Comment:
We should consider the case for `reader.onerror` also.

Reply:
Hello @shsa-odoo  ,

Currently, if a profile picture upload fails, an error is already displayed upon saving. Are we expected to test this functionality twice? Additionally, occurrences of broken uploads or files are rare, so adding further complexity might not be necessary.

Reply:
Hey @panv-odoo After discussing with SHRA and @msh-odoo  about the failed loading of image rather than traceback which we get to see after saving the whole information , its better to showcase the toaster notification as we do for the other image based snippets.

And for your current code you should handle the `onerror` case where you should simply set the src to the default one or empty it. which will prevent causing the traceback when we save the information, resulting in a better UX.